labgrid/driver/sshdriver.py

```diff
@@ -118,10 +126,46 @@ def _run(self, cmd, codec="utf-8", decodeerrors="strict", timeout=None): # pylin
         returns:
         (stdout, stderr, returncode)
         """
+        return self._run(cmd, codec=codec, decodeerrors=decodeerrors)
+
+    def _run(self, cmd, codec="utf-8", decodeerrors="strict", timeout=None): # pylint: disable=unused-argument
+        timeout_ = None if timeout is None else Timeout(float(timeout))
+
+        process = self._run_as_background_process(cmd)
+        read_timeout = timeout_.remaining if timeout_ else None
+        stdout, stderr = self._read_from_background_process(process, timeout=read_timeout)
+
+        if timeout_ and timeout_.expired:
```

Comment:
Also check if process is meanwhile finished here like in the shell driver
```suggestion
        if timeout_ and timeout_.expired and process.poll():
```